cli/docker_validator.py

```diff
@@ -0,0 +1,227 @@
+
+
+def run_command_with_output(command_description, command, return_process_output=True,
+                            log_error_stream_to_output=False):
+    process_output = ""
+
+    logging.info(f"Starting process: {command_description}")
+    logging.info(f"Running command: {command}")
+
+    stdout = subprocess.PIPE
+    # Some utilities output non-error messages to error stream. This is a workaround for that
+    stderr = subprocess.STDOUT if log_error_stream_to_output else subprocess.PIPE
+
+    with subprocess.Popen(command, stdout=stdout, stderr=stderr, bufsize=1, universal_newlines=True,
+                          shell=True) as process:
+        for line in iter(process.stdout.readline, ''):
+            line = str(line).rstrip()
+            logging.info(line)
+            if return_process_output:
+                process_output += line + "\n"
+        if not log_error_stream_to_output:
+            for line in iter(process.stderr.readline, ''):
+                line = str(line).rstrip()
+                logging.error(line)
+    if process.returncode != 0:
+        logging.error(f"{command_description} - failed! Refer to the error messages for details.")
+        raise subprocess.CalledProcessError(process.returncode, process.args)
+    else:
+        logging.info(f"{command_description} - completed successfully")
+    if return_process_output:
+        return process_output
+
+
+def check_if_file_missing(mapping_file):
+    files_missing = False
+    missing_files_list = []
+    with open(mapping_file) as open_file:
+        reader = csv.DictReader(open_file, delimiter=',')
+        for row in reader:
+            if not os.path.exists(row['vcf']):
+                files_missing = True
+                missing_files_list.append(row['vcf'])
+            if not os.path.exists(row['fasta']):
+                files_missing = True
+                missing_files_list.append(row['fasta'])
+            if not os.path.exists(row['report']):
+                files_missing = True
+                missing_files_list.append(row['report'])
+
+    return files_missing, missing_files_list
+
+
+def verify_docker_is_installed(docker):
+    try:
+        run_command_with_output("check docker is installed and available on the path", f"{docker} --version")
+    except subprocess.CalledProcessError as ex:
+        logging.error(ex)
+        raise RuntimeError(f"Please make sure docker ({docker}) is installed and available on the path")
+
+
+def verify_container_is_running(docker, container_name):
+    container_run_cmd_ouptut = run_command_with_output("check if container is running", f"{docker} ps")
+    if container_run_cmd_ouptut is not None and container_name in container_run_cmd_ouptut:
+        logging.info(f"Container ({container_name}) is running")
+        return True
+    else:
+        logging.info(f"Container ({container_name}) is not running")
+        return False
+
+
+def verify_container_is_stopped(docker, container_name):
```

Comment:
The fact that your passing `docker` and `container_name` around makes me think that this should be a class.